module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,228 @@
+                return;
+            }
+
+            // Deserialise the message into a neutron
+            SensorMeasurements msg = NUClear::util::serialise::Serialise<SensorMeasurements>::deserialise(data);
+
+            // Read each field of msg, translate it to our protobuf and emit the data
+            auto sensor_data = std::make_unique<DarwinSensors>();
+            sensor_data.FromSeconds(msg.time);
+
+            for (auto& position : msg.position_sensor) {
+                // string name
+                // double valvue
+            }
+
+            for (auto& accelerometer : msg.accelerometer) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& gyro : msg.gyro) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& bumper : msg.bumper) {
+                // string name
+                // bool value
+            }
+
+            for (auto& force_3d : msg.force_3d) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& force_6d : msg.force_6d) {
+                // string name
+                // Vector3 value
+            }
+
+            emit(sensor_data);
+
+            for (auto& camera : msg.camera) {
+                // Convert the incoming image so we can emit it to the PowerPlant.
+                auto compressed_image = std::make_unique<CompressedImage>();
+                compressed_image->timestamp.FromSeconds(msg.time);
+                compressed_image->name            = camera.name;
+                compressed_image->dimensions.x()  = camera.width;
+                compressed_image->dimensions.y()  = camera.height;
+                compressed_image->format          = camera.quality;  // This is probably wrong, we havent documented :(
+                compressed_image->data            = camera.data;
+                compressed_image->lens.fov        = (float) camera.horizontalFieldOfView;
+                compressed_image->lens.centre.x() = (float) camera.centerX;
+                compressed_image->lens.centre.y() = (float) camera.centerY;
+                // Radial coefficients
+                // tangential coefficients
+                emit(compressed_image);
+            }
+        });
+
+        on<Every<1, std::chrono::seconds>>().then([this, fd]() {
+            // Sending
+            std::vector<char> data = NUClear::util::serialise::Serialise<ActingMessage>::serialise(to_send);
+            uint64_t N             = data.size();
+            send(fd, &N, sizeof(N), 0);
+            send(fd, data.data(), N, 0);
+        });
+    });
+
+    // Create the message that we are going to send.
+    on<Trigger<ServoTargets>>().then([this](const ServoTargets& commands) {
+        // Maybe keep the `ServoTarget`s we have not sent yet, instead of just overriding them.
+        to_send = ActuatorRequests();
```

Comment:
You probably want to go through each element and clear the vectors.
```suggestion
        to_send.motor_position.clear();
        to_send.motor_velocity.clear();
        to_send.motor_torque.clear();
```